graveler/branch_locker_test.go

```diff
@@ -10,73 +13,153 @@ import (
 )
 
 func TestBranchLock(t *testing.T) {
-	bl := graveler.NewBranchLocker()
+	conn, _ := tu.GetDB(t, databaseURI)
+	bl := graveler.NewBranchLocker(conn)
 
-	closeWrite, err := bl.AquireWrite("a", testutil.DefaultBranchID)
-	tu.MustDo(t, "acquire write I", err)
-	closeWrite()
+	ctx := context.Background()
+	t.Run("multiple_writers", func(t *testing.T) {
+		stopWritersCh := make(chan struct{})
+		const writers = 5
+		var wgLocked sync.WaitGroup
+		var wgDone sync.WaitGroup
+		wgLocked.Add(writers)
+		wgDone.Add(writers)
+		for i := 0; i < writers; i++ {
+			go func() {
+				defer wgDone.Done()
+				_, err := bl.Writer(ctx, "a", testutil.DefaultBranchID, func() (interface{}, error) {
+					wgLocked.Done()
```

Comment:
Consider waiting for all the other writers to lock here (i.e. add _another_ WaitGroup, add 1 to it before calling `bl.Writer` and calling its Done and then its Wait in here.  Otherwise the scheduler could just run the writers in sequence :-(